dev/AppLifecycle/ActivationRegistrationManager.cpp

```diff
@@ -8,47 +8,89 @@
 #include "ProtocolActivatedEventArgs.h"
 #include "FileActivatedEventArgs.h"
 #include "Association.h"
+#include "ExtensionContract.h"
+#include "EncodedLaunchExecuteCommand.h"
+
+extern "C" HRESULT __stdcall DllRegisterServer() noexcept;
 
 namespace winrt::Microsoft::ProjectReunion::implementation
 {
+    std::wstring GenerateEncodedLaunchUri(std::wstring const& appUserModelId, std::wstring const& contractId)
+    {
+        return c_encodedLaunchSchemeName + L"://" + appUserModelId + L"?ContractId=" + contractId;
+    }
+
+    std::wstring GenerateCommandLine(std::wstring const& modulePath)
+    {
+        std::wstring command = modulePath;
+
+        if (modulePath.empty())
+        {
+            command = GetModulePath();
+        }
+
+        command += L" " + c_argumentPrefix + c_protocolArgumentString + c_argumentSuffix;
```

Comment:
> command += L" " + c_argumentPrefix + c_protocolArgumentString + c_argumentSuffix; [](start = 7, length = 82)

This causes multiple reallocations and copies. Is there a more efficient option e.g. 

```
auto command{ std::wstring::concat(modulePath.empty() ? GetModulePath() : moduelPath, L" ", c_argumentPrefix, c_protocolArgumentString, c_argumentSuffix) };
```

Reply:
Small performance cost for readability and ease of use.